packages/components/src/components/domainproperties/dataset/models.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2020 LabKey Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Record} from "immutable";
+import {DomainDesign} from "../models";
+
+
+export class DatasetModel extends Record({
+    domain: undefined,
+    entityId : undefined,
+    createdBy : undefined,
+    created : undefined,
+    modifiedBy : undefined,
+    modified : undefined,
+    containerId : undefined,
+    datasetId: undefined,
+    name: undefined,
+    typeURI: undefined,
+    category: undefined,
+    categoryId: undefined,
+    visitDatePropertyName: undefined,
+    keyProperty: undefined,
+    isDemographicData: undefined,
+    label: undefined,
+    cohortId: undefined,
+    tag: undefined,
+    showInOverview: undefined,
+    description: undefined,
+    sourceAssayName: undefined,
+    sourceAssayURL: undefined,
+    dataSharing: undefined,
+    keyManagementType: undefined
+}) {
+    domain: DomainDesign;
+    datasetId?: number;
+    name: string;
+    typeURI?: string;
+    category?: string;
+    categoryId?: number;
+    visitDatePropertyName?: string;
+    keyProperty?: string;
+    isDemographicData?: boolean;
+    label?: string;
+    cohortId?: number;
+    tag?: string;
+    showInOverview?: boolean;
+    description?: string;
+    sourceAssayName?: string;
+    sourceAssayURL?: string;
+    dataSharing?: string;
+    keyManagementType?: any;
```

Comment:
Any reason so many of these properties are optional? I would have thought they would all be defined without the `?`.